pkg/actions/nodegroup/upgrade.go

```diff
@@ -46,19 +50,39 @@ func (m *Manager) upgradeAndWait(nodeGroupName, version, launchTemplateVersion s
 		Version:       &version,
 	}
 
+	describeNodegroupOutput, err := m.ctl.Provider.EKS().DescribeNodegroup(&eks.DescribeNodegroupInput{
+		ClusterName:   &m.cfg.Metadata.Name,
+		NodegroupName: &nodeGroupName,
+	})
+
+	if err != nil {
+		return err
+	}
+
 	if launchTemplateVersion != "" {
-		describeNodegroupOutput, err := m.ctl.Provider.EKS().DescribeNodegroup(&eks.DescribeNodegroupInput{
-			ClusterName:   &m.cfg.Metadata.Name,
-			NodegroupName: &nodeGroupName,
-		})
-		if err != nil {
-			return err
+		lt := describeNodegroupOutput.Nodegroup.LaunchTemplate
+		if lt != nil && lt.Id != nil {
+			input.LaunchTemplate = &eks.LaunchTemplateSpecification{
+				Version: &launchTemplateVersion,
+				Id:      describeNodegroupOutput.Nodegroup.LaunchTemplate.Id,
+			}
+		} else if lt != nil && lt.Name != nil {
+			input.LaunchTemplate = &eks.LaunchTemplateSpecification{
+				Version: &launchTemplateVersion,
+				Name:    describeNodegroupOutput.Nodegroup.LaunchTemplate.Name,
+			}
+		} else {
+			return fmt.Errorf("cannot update launch template version because the nodegroup is not configured to use one")
 		}
```

Comment:
Very minor nit (just something I prefer) that can be ignored:

```suggestion
		if lt == nil || lt.Id == nil || lt.Name == nil {
			return errors.New("cannot update launch template version because the nodegroup is not configured to use one")
		}
		input.LaunchTemplate = &eks.LaunchTemplateSpecification{
			Version: &launchTemplateVersion,
			Id:      describeNodegroupOutput.Nodegroup.LaunchTemplate.Id,
			Name:    describeNodegroupOutput.Nodegroup.LaunchTemplate.Name,
		}
```

Reply:
I like! 😄

Reply:
I like the refactor, but I think only one of ID or name need to be set?

Reply:
Yes, that makes sense.